sonar-plugin-api/src/main/java/org/sonar/api/component/SourceFile.java

```diff
@@ -19,5 +19,9 @@
  */
 package org.sonar.api.component;
 
+/**
+ * @deprecated since 5.6
+ */
+@Deprecated
 public interface SourceFile extends Component {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)